File: capstone/vue/src/services/MealService.js
import axios from 'axios';

export default{
    getMeals(){
        return axios.get('/meals')
    },

    getMealById(id){
        return axios.get(`/meals/${id}`)
    },

    addMeal(meal){
        return axios.post('/meals', meal)
    },

    updateMeal(meal){
        return axios.put(`/meals/${meal.id}`, meal)
    },

    deleteMeal(id){
        return axios.delete`/meals/${id}`
    },

    addRecipeToMeal(mealId, recipeId){
        return axios.post(`/meals/${mealId}/modify/${recipeId}`, null, 
        {params: {mealId, recipeId}
    });
    }
}

File: capstone/java/src/main/java/com/techelevator/service/MealService.java
package com.techelevator.service;

import com.techelevator.model.Meal;

import java.security.Principal;
import java.util.List;

public interface MealService {
    List<Meal> getMeals(Principal user) throws InterruptedException;
    Meal getMealById(int id, Principal user) throws InterruptedException;
    Meal createMeal(Meal meal, Principal user);
    Meal updateMealInfo(int id, Meal updatedMeal, Principal user);
    int addRecipeToMeal (int mealId, int recipeId, Principal user);
    int removeRecipeFromMeal(int mealId, int recipeId, Principal user);

}
